storybook/stories/API/component/Legend.stories.tsx

```diff
@@ -36,3 +36,66 @@ export const API = {
   },
   args: getStoryArgsFromArgsTypesObject(LegendProps),
 };
+
+export const LegendPortal = {
+  render: (args: Args) => {
+    const [, surfaceHeight] = [600, 300];
+    const [legendPortal, setLegendPortal] = useState<HTMLDivElement | null>(null);
+
+    return (
+      <div style={{ display: 'flex', width: '100%', height: surfaceHeight }}>
+        <div
+          style={{ width: '100%', height: '100%' }}
+          ref={node => {
+            if (legendPortal == null && node != null) {
+              setLegendPortal(node);
+            }
+          }}
+        >
+          <h2>Recharts Downloads</h2>
+          We can render whatever we want render with the Legend items. This chart represents Recharts downloads from
+          August 2023 to August 2024.
+        </div>
+        <div style={{ width: '100%', height: '100%' }}>
+          <ResponsiveContainer width="100%" height={surfaceHeight}>
+            <AreaChart
+              margin={{
+                top: 20,
+                right: 20,
+                bottom: 20,
+                left: 20,
+              }}
+              data={rechartsPackageDownloads}
+            >
+              {/* The target component */}
+              {legendPortal && (
+                <Legend
+                  {...args}
+                  // TODO: when a portal is passed remove absolute positioning and other styles
```

Comment:
When should we fix this TODO? In this PR? Or before releasing 3.x?

Reply:
Didn't want to do it in this PR - at this point it's at the same place as Tooltip. I can take both Tooltip and this next thing I do